### PR TITLE
Fix yang validation failure when table contains empty value

### PR DESCRIPTION
**Pull request by wen587**
#### Why I did it
Fix https://github.com/Azure/sonic-buildimage/issues/9746
#### How I did it
Split the check condition based on non-exist and zero length.
#### How to verify it
Run verification script when table contains empty value
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: src/sonic-yang-mgmt/sonic_yang_ext.py [hunk 1]
```diff
@@ -593,7 +593,11 @@ def _xlateContainerInContainer(self, model, yang, configC, table):
         ccontainer = model
         #print(ccontainer['@name'])
         yang[ccontainer['@name']] = dict()
-        if not configC.get(ccontainer['@name']):
+        if configC.get(ccontainer['@name']) is None:
```
Comment (qiluo-msft) [inline src/sonic-yang-mgmt/sonic_yang_ext.py:596]: >configC.get(ccontainer['@name']) is None [](http://example.com/codeflow?start=11&length=40)

How about
```
if ccontainer['@name'] not in configC:
```

BTW, you can extract `ccontainer['@name']` as a variable, no need to calculate many times.

And remove `#print` above.

**Review by qiluo-msft — COMMENTED** [review]
LGTM

File: src/sonic-yang-mgmt/sonic_yang_ext.py [hunk 2]
```diff
@@ -591,18 +591,22 @@ def _xlateListInContainer(self, model, yang, configC, table, exceptionList):
...
-        yang[ccontainer['@name']] = dict()
-        if not configC.get(ccontainer['@name']):
+        ccName = ccontainer['@name']
+        yang[ccName] = dict()
+        if ccName not in configC:
             return
-        self.sysLog(msg="xlateProcessListOfContainer: {}".format(ccontainer['@name']))
-        self._xlateContainer(ccontainer, yang[ccontainer['@name']], \
-        configC[ccontainer['@name']], table)
+        if len(ccName) == 0:
```
Comment (praveen-li) [inline src/sonic-yang-mgmt/sonic_yang_ext.py:598]: if len(configC[ccName]) ==0: 

i.e if value is zero, delete the key. Here, delete the key means processing is done,

**Review by praveen-li — COMMENTED** [review]
One Dev Changes is needed.

Will it be possible for you to add a test case for this please, may be in same reference under which you found the issue. Or simply by adding a config in https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/tests/files/sample_config_db.json

File: src/sonic-yang-mgmt/sonic_yang_ext.py (hunk 2 above)
Comment (wen587) [reply src/sonic-yang-mgmt/sonic_yang_ext.py]: Hi, I saw TACPLUS table already has configuration in sample_config_db.json.
I added a UT for field `global` with empty value.
From where I saw the issue, it failed at GCU test after other test configured TACPLUS table.
I also added a empty json apply GCU test https://github.com/Azure/sonic-mgmt/pull/5459  for simply checking the configDB.

File: src/sonic-yang-models/tests/files/sample_config_db.json [hunk 3]
```diff
@@ -304,7 +304,7 @@
                 "switch_id": "2",
                 "switch_type": "voq",
                 "max_cores": "8",
-                "sub_role": "FrondEnd",
```
Comment (praveen-li) [inline src/sonic-yang-models/tests/files/sample_config_db.json:307]: [10:18 PM] Praveen Chaudhary
in this file: src/sonic-yang-models/tests/files/sample_config_db.json

[10:18 PM] Praveen Chaudhary
we need global {} config for test to exercise this code 

[10:18 PM] Praveen Chaudhary
plz add that and we are good

**Review by praveen-li — APPROVED** [review]
LGTM

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**